### DISCUSSION TITLE
Matter Air Quality Sensor with MHZ19B not showing values in Apple Home

### DISCUSSION THREAD

**Discussion by stony2k** [arendst/Tasmota#22701] [Q&A] [answered]
I'm trying to get my MHZ19B CO2 sensor working with Matter in Tasmota (14.4.1) and Apple Home. While the sensor itself works correctly and Matter is configured, Apple Home shows the air quality as "Unknown".

### Current Setup
- Tasmota 14.4.1
- MHZ19B CO2 sensor (working correctly)
- Matter enabled and paired with Apple Home
- SetOption134 1 (Matter debug enabled)

### What I've tried
1. Configured Matter endpoint for Air Quality Sensor with `MHZ19B#CarbonDioxide`
2. Sensor delivers correct values:
14:03:05.045 MQT: tele/tasmota_B8CCBC/SENSOR = {"Time":"2024-12-22T14:03:05","MHZ19B":{"Model":"B","CarbonDioxide":908,"Temperature":32.0},"TempUnit":"C"}

### Troubleshooting Steps
- Factory reset Matter
- Removing and re-adding the device in Apple Home
- Different Matter endpoint configurations
- Restarting Tasmota

### Related Code
- [Matter Air Quality Sensor Implementation](https://github.com/arendst/Tasmota/blob/master/lib/libesp32/berry_matter/src/embedded/Matter_Plugin_2_Sensor_Air_Quality.be)
- [Pull Request that added Air Quality support](https://github.com/arendst/Tasmota/pull/21559/commits/28fd3bbbf99e5c2d3d0b918a71e38b8c1011e38c)

### Question
What am I missing to get the CO2 values showing up correctly in Apple Home? The sensor works, Matter is configured, but the values are not being transmitted correctly. According to the berry script implementation, the values should be transmitted in a correct format to Apple Home.

Do I need to manually import the script, or is there something else I'm missing in the configuration?

Any help would be greatly appreciated!

**Comment by s-hadinger** [comment]
Before starting to debug, I'm not sure that Apple already supports Air Quality Sensors in Matter. Did you have confirmation that iOS 18.1 or 18.2 supports this type of device?

**Comment by stony2k** [comment, marked as answer]
I posted some information in the issue: https://github.com/arendst/Tasmota/issues/22704#issuecomment-2558538258

Should I close this discussion while the issue is open and ongoing?
